### PR TITLE
Enable Requester Pays Buckets

### PR DESCRIPTION
**Pull request by 3DTOPO**
#### Relevant Issue (if applicable)
_If there are Issues related to this PullRequest, please list it._

#### Details
_Please describe the details of PullRequest._

### CODE REVIEW

File: src/curl.cpp [hunk 1]
```diff
@@ -2276,6 +2276,7 @@ void S3fsCurl::insertV4Headers(const string &op, const string &path, const strin
   requestHeaders = curl_slist_sort_insert(requestHeaders, "host", get_bucket_host().c_str());
   requestHeaders = curl_slist_sort_insert(requestHeaders, "x-amz-content-sha256", contentSHA256.c_str());
   requestHeaders = curl_slist_sort_insert(requestHeaders, "x-amz-date", date8601.c_str());
+  requestHeaders = curl_slist_sort_insert(requestHeaders, "x-amz-request-payer", "requester");
```
Comment (andrewgaul) [inline src/curl.cpp]: Why do you add this header unconditionally?

Comment (3DTOPO) [reply src/curl.cpp]: Because it fixed the issue I was having, and as I stated, it would ideally be added as an option

Comment (andrewgaul) [reply src/curl.cpp]: Sending unneeded headers to some S3 implementations will cause requests to fail, as with S3Proxy running in Travis.  Please add a flag.

**Comment by ggtakec** [comment]
@3DTOPO 
You can not add the x-amz-request-payer header for every request.
Instead of it, you can prepare a dedicated startup option and add a header only when you specify that option.

I think this method is good.
And I reopened your #635 issue, thus we need to discuss about request payer.

Regards,

**Comment by ggtakec** [comment]
@3DTOPO This PR has been left unmodified.
Since we have made considerable changes from the base code, I will close this PR.
Please open again if there is code update.
Thanks in advance for your assistance.

File: src/curl.cpp (hunk 1 above)
Comment (ahuarte47) [reply src/curl.cpp]: Hi, I have created this [PR](https://github.com/s3fs-fuse/s3fs-fuse/pull/1199) implementing this flag as a new option. I hope this helps!

**Pull request closed without merging**